front/lib/api/gens.ts

```diff
@@ -0,0 +1,102 @@
+import { Op } from "sequelize";
+
+import { GensTemplateType } from "@app/types/gens";
+import { UserType, WorkspaceType } from "@app/types/user";
+
+import { GensTemplate } from "../models";
+
+export async function getGensTemplates(
+  owner: WorkspaceType,
+  user: UserType
+): Promise<GensTemplateType[]> {
+  const templates = await GensTemplate.findAll({
+    where: {
+      workspaceId: owner.id,
+      [Op.or]: [
+        {
+          visibility: "workspace",
+        },
+        {
+          userId: user.id,
+        },
+      ],
+    },
+    order: [["createdAt", "DESC"]],
+  });
+
+  return templates.map((t) => {
+    return {
+      name: t.name,
+      instructions: t.instructions,
+      sId: t.sId,
+      color: t.color,
+      userId: t.userId,
+      visibility: t.visibility,
+    };
+  });
+}
+
+export async function getTemplate(
+  owner: WorkspaceType,
+  user: UserType,
+  sId: string
+): Promise<GensTemplateType | null> {
+  const template = await GensTemplate.findOne({
+    where: {
+      workspaceId: owner.id,
+      [Op.or]: [
```

Comment:
I think we can remove that or entirely and we can remove the user argument no?

Reply:
then you could get another user's `user` template, is that ok?

Reply:
You need the templateId which is kind of hard to get but fine to leave it here 👍